### PR TITLE
[#28] refactor to reduce code repetition

### LINKED ISSUE #28: refactor to reduce code repetition
There are a lot of methods that have similar code blocks. They can be extracted and reuse.

### PR DESCRIPTION
**Pull request by gaohoward**
Currently in ArtemisJolokia code there are many common codes among
fetching methods for different components. For example
readBrokerAttributes, readAddressAttributes and
readClusterConnectionAttributes are all sending 'POST' requests
with similar pattern but only different parameters.

It can be abstracted to a common method shared by those methods
and also will be reused when we add new components like Bridges.

### CODE REVIEW

File: src/api/apiutil/artemis_jolokia.ts [hunk 1]
```diff
@@ -648,21 +303,53 @@ export class ArtemisJolokia {
...
+  getComponentNameKey(compType: ComponentType): string {
+    switch (compType) {
+      case ComponentType.ACCEPTOR: {
+        return 'ACCEPTOR_NAME';
+      }
+      case ComponentType.ADDRESS: {
+        return 'ADDRESS_NAME';
+      }
+      case ComponentType.BROKER: {
+        return '';
```
Comment (Msarawan) [inline src/api/apiutil/artemis_jolokia.ts:315]: Hello @gaohoward ,
here it should return 'BROKER_NAME' right?? instead of ' ' string...

**Comment by lavocatt** [comment]
Can you add more context to the PR and the commit please?

File: src/api/apiutil/artemis_jolokia.ts (hunk 1 above)
Comment (gaohoward) [reply src/api/apiutil/artemis_jolokia.ts:315]: Hi Marjina, this is a special case for `BROKER` component. We use this method to find the key to replace component names. The `BROKER` component is not used here as we always search for it to replace for each component (each component query will have `BROKER_NAME` search and replace. Here right before calling this method:
https://github.com/gaohoward/activemq-artemis-jolokia-api-server/blob/j_qaddr_ops/src/api/apiutil/artemis_jolokia.ts#L341

**Comment by gaohoward** [comment]
> Can you add more context to the PR and the commit please?

Sure will do. Thanks!

File: src/api/apiutil/artemis_jolokia.ts (hunk 1 above)
Comment (Msarawan) [reply src/api/apiutil/artemis_jolokia.ts:315]: yeah, got it. thanks!

**Review by Msarawan — APPROVED** [review]
LGTM!

**Review by lavocatt — APPROVED** [review]
Thanks for having added the context to the PR.

**Pull request merged**